.cargo/config.toml

```diff
@@ -1,5 +1,58 @@
-# Deployment runtime lints
-[target.'cfg(all(target_arch = "wasm32", not(test)))']
+# Unfortunately we cannot add any lints based on `cfg(test)` nor `cfg(feature = "feat")`: https://github.com/rust-lang/cargo/issues/8170.
+# Hence, we can only define lints that are applied codebase-wide (including tests), and lints that are applied only to runtime code.
+
+# Codebase-wide lints.
+[target.'cfg(all())']
+rustflags = [
+  # We need to deny warnings here instead of with the `RUSTFLAGS` env since the env variable would completely override these settings -> https://github.com/rust-lang/cargo/issues/5376.
+  "-Dwarnings",
+  "-Wclippy::as_underscore",
+  "-Wclippy::branches_sharing_code",
+  "-Wclippy::clear_with_drain",
+  "-Wclippy::clone_on_ref_ptr",
+  "-Wclippy::collection_is_never_read",
+  "-Wclippy::derive_partial_eq_without_eq",
+  "-Wclippy::else_if_without_else",
+  "-Wclippy::empty_drop",
+  "-Wclippy::empty_structs_with_brackets",
```

Comment:
We should do the same for enums: [empty_enum_variants_with_brackets ](https://rust-lang.github.io/rust-clippy/master/index.html#empty_enum_variants_with_brackets)

Reply:
Only available in 1.77 onwards.

Reply:
Plus it's already commented in the config file, so it just needs to be uncommented once we upgrade the compiler version.

Reply:
Oh yes, I somehow missed that